features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/DefaultSystemRequirement.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.prerequisite
+
+import android.os.Handler
+import android.os.Looper
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.utils.submitSafe
+import com.datadog.android.sessionreplay.SystemRequirementConfiguration
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+internal class DefaultSystemRequirement(
+    private val executorService: ExecutorService = Executors.newSingleThreadExecutor(),
+    private val handler: Handler = Handler(Looper.getMainLooper())
+) : SystemRequirement {
+
+    override fun runIfMeetRequirements(
+        systemRequirementConfiguration: SystemRequirementConfiguration,
+        checkers: List<SystemRequirementChecker>,
+        internalLogger: InternalLogger,
+        runnable: () -> Unit
+    ) {
+        executorService.submitSafe(OPERATION_NAME, internalLogger) {
+            if (checkAllMinimumRequirements(systemRequirementConfiguration, checkers)) {
+                handler.post {
+                    runnable()
+                }
+            } else {
+                internalLogger.log(
+                    level = InternalLogger.Level.INFO,
+                    listOf(InternalLogger.Target.TELEMETRY, InternalLogger.Target.USER),
+                    messageBuilder = {
+                        "Session replay is disabled because the system doesn't meet the minimum SDK requirements"
```

Comment:
I would remove the word 'SDK' to prevent customer confusion. Here we are setting minimum requirements for Session Replay, but the rest of our SDK continues to work normally, but if we have the word 'SDK' then we may imply that this is also the minimum requirements for Logs, Traces, etc.